mdbook-slide-evaluator/src/evaluator.rs

```diff
@@ -0,0 +1,222 @@
+    /// store screenshot in this directory if provided
+    screenshot_dir: Option<PathBuf>,
+    /// html base uri to the source_dir used as a prefix for each page
+    html_base_url: Url,
+    /// base directory for all processed files
+    source_dir: PathBuf,
+}
+
+/// element coordinates returned by the browser
+#[derive(Debug)]
+struct ElementSize {
+    /// the width of the element
+    width: f64,
+    /// the height of the element
+    height: f64,
+}
+
+impl From<(f64, f64, f64, f64)> for ElementSize {
+    fn from(value: (f64, f64, f64, f64)) -> Self {
+        Self { width: value.2, height: value.3 }
+    }
+}
+
+#[derive(Debug)]
+/// holds the evaluation result for a slide
+pub struct EvaluationResult {
+    /// metadata about the slide
+    slide: Slide,
+    /// the size of the main content element
+    element_size: ElementSize,
+}
+
+/// holds all evaluation results for a book
+pub struct EvaluationResults {
+    /// metadata about the book
+    book: Book,
+    /// the collected evaluation results
+    results: Vec<EvaluationResult>,
+}
+
+#[derive(Serialize)]
+struct ExportFormat {
+    filename: PathBuf,
+    element_width: usize,
+    element_height: usize,
+}
+
+impl EvaluationResults {
+    /// export the evaluation results to the given csv file, overwrites if
+    /// allowed
+    pub fn export_csv(&self, file: &Path, overwrite: bool) -> anyhow::Result<()> {
+        if file.exists() && !overwrite {
+            Err(anyhow!(
+                "Not allowed to overwrite existing evaluation results at {}",
+                file.display()
+            ))?;
+        };
+
+        let mut csv_writer = csv::Writer::from_path(file)?;
+        for result in &self.results {
+            csv_writer.serialize(ExportFormat {
+                filename: (*result.slide.filename).to_path_buf(),
+                element_width: result.element_size.width.round() as usize,
+                element_height: result.element_size.height.round() as usize,
+            })?;
+        }
+        Ok(())
+    }
+
+    /// dump the results to stdout
+    pub fn export_stdout(&self) {
+        for result in &self.results {
+            println!(
+                "{}: {}x{}",
+                result.slide.filename.display(),
+                result.element_size.width,
+                result.element_size.height
+            );
+        }
+    }
+}
+
+impl<'a> Evaluator<'_> {
+    /// create a new instance with the provided config.
+    /// fails if the webclient cannot be created
+    pub async fn new(
+        webdriver: &str,
+        element_selector: &'a str,
+        screenshot_dir: Option<PathBuf>,
+        html_base_url: Url,
+        source_dir: PathBuf,
+    ) -> anyhow::Result<Evaluator<'a>> {
+        let webclient =
+            fantoccini::ClientBuilder::native().connect(webdriver).await?;
+        // use fullscreen window to avoid arbitrary window size limitations
+        webclient.fullscreen_window().await?;
```

Comment:
Is this a well-known size? What size is it?

Reply:
A very good question and short answer: it depends.

It could be 1920x1080 if e.g. docker-selenium containers are used: https://github.com/SeleniumHQ/docker-selenium/blob/68ebfe1a39a533dfbc4b3fe37701bcfbff6da005/NodeBase/Dockerfile#L53
Setting this to a fixed (configurable) value will provide a more repeatable experience